domain/machine/state/reboot.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+
+	"github.com/canonical/sqlair"
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/internal/database"
+)
+
+// RebootAction defines the action a machine should
+// take when a hook needs to reboot
+type RebootAction int
+
+const (
+	// ShouldDoNothing instructs a machine agent that no action
+	// is required on its part
+	ShouldDoNothing RebootAction = iota
```

Comment:
**issue:** We're moving away from `iota`, only because there is a gotcha if you accidentally add something in the middle of the list.

**suggestion**: Move this to a string, it's more descriptive and doesn't cost us, as correctness is more important.